src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+import React from 'react';
+import classnames from 'classnames';
+import Icon from '../Icon';
+import checkboxStyles from './checkbox.scss';
+
+export default class Checkbox extends React.PureComponent {
+    props: {
+        checked: boolean,
+        skin: 'dark' | 'light',
+        className: string,
+        onChange?: (checked: boolean) => void,
+    };
+
+    input: HTMLInputElement;
+
+    static defaultProps = {
+        skin: 'dark',
+        className: '',
+    };
+
+    handleChange = (event: Event) => {
+        if (this.props.onChange && typeof event.target.checked === 'boolean') {
```

Comment:
Maybe it works if you use the correct type annotation for `event`?  Apart from that you should use the [SyntheticEvent](https://flow.org/en/docs/react/events/).

There you can also pass a template:

```javascript
handleChange = (event: SyntheticEvent<HTMLInputElement>) => { /* ... */ };
```

Maybe the boolean check is not needed anymore afterwards.